src/sync/app.cpp

```diff
@@ -87,9 +89,32 @@ const static uint64_t    default_timeout_ms = 60000;
 const static std::string username_password_provider_key = "local-userpass";
 const static std::string user_api_key_provider_key_path = "api_keys";
 
-SharedApp App::get_shared_app(const Config& config)
+SharedApp App::get_shared_app(const Config& config, const SyncClientConfig& sync_client_config)
```

Comment:
All of these functions which access the cache need locking.

Reply:
Done